### DISCUSSION TITLE
How do updating/updated lifecycle hooks work with arrays?

### DISCUSSION THREAD

**Discussion by dakira** [livewire/livewire#7264] [1. Help]
Hi,

I have a problem with some code that worked in Livewire v2 but doesn't in v3 because arrays are handled differently by the lifecycle hooks.

Imagine this component

```php
class Roles extends Component
{
    public array $rolesSelected = [];
}
```

and a view

```html
<input type="checkbox" wire:model.live="rolesSelected" value="1"/> Reader
<input type="checkbox" wire:model.live="rolesSelected" value="2"/> Editor
<input type="checkbox" wire:model.live="rolesSelected" value="3"/> Admin
<input type="checkbox" wire:model.live="rolesSelected" value="4"/> Superuser
```

When I check Reader and Editor, `$rolesSelected` contains `[1, 2]`, just as expected.

When I add an updating hook, it gets weird, though:

```
public function updatingRolesSelected($value)
{
    \Log::debug($value);
}
```

Given a state where all 4 roles are selected (`[1, 2, 3, 4]`), I deselect the first role. Here's what happens:

###  Livewire v2
The hook fires once and `$value` contains the state that $this->rolesSelected will have after the update: `[2, 3, 4]`.

### Livewire v3
The hook fires four times. Once for every element of the array. `$value` contains 2, 3, 4 and '_rm_'.

Watching `$this->rolesSelected` while the hooks are fired, it looks like this:
- `[1, 2, 3, 4]`
- `[2, 2, 3, 4]`
- `[2, 3, 3, 4]`
- `[2, 3, 4, 4]`

So is there any way to have something like the Livewire v2 behavior (i.e. only one hook fired for the array)?

**Comment by iamnoorsk** [comment]
i'm facing similar issues
![image](https://github.com/livewire/livewire/assets/89458293/df6bfb6a-974c-47f5-9191-728409c217ea)

**Comment by iamnoorsk** [comment]
@dakira check this discussion
https://github.com/livewire/livewire/discussions/6665
